### ISSUE TITLE
Crash in IBGMethodsLogger LogMethod:

### ISSUE THREAD

**Issue by lha-ryq** [Instabug/Instabug-iOS#169]
We see new crash related with Instabug. It looks like it happens after calling setUserData.
Instabug version: 7.2.6

```
-[NSComparisonPredicate rangeOfString:]: unrecognized selector sent to instance 0x170240990

0   CoreFoundation                  __exceptionPreprocess + 132
1   libobjc.A.dylib                 objc_exception_throw + 56
2   CoreFoundation                  __methodDescriptionForSelector + 0
3   CoreFoundation                  ___forwarding___ + 924
4   CoreFoundation                  __forwarding_prep_0___ + 124
5   Instabug                        -[IBGMethodsLogger LogMethod:] (IBGMethodsLogger.m:71)
6   Instabug                        __24+[Instabug setUserData:]_block_invoke (Instabug.m:156)
7   Foundation                      __24-[NSBlockOperation main]_block_invoke + 8
8   Foundation                      -[NSBlockOperation main] + 112
9   Foundation                      -[__NSOperationInternal _start:] + 628
10  Foundation                      __NSOQSchedule_f + 244
11  libdispatch.dylib               _dispatch_client_callout + 16
12  libdispatch.dylib               _dispatch_queue_serial_drain + 928
13  libdispatch.dylib               _dispatch_queue_invoke + 884
14  libdispatch.dylib               _dispatch_root_queue_drain + 540
15  libdispatch.dylib               _dispatch_worker_thread3 + 124
16  libsystem_pthread.dylib         _pthread_wqthread + 888
17  libsystem_pthread.dylib         0x18eae6cac
```

Here is the execution of `Instabug.setUserData` in our code:
```
let username = AuthenticationCache.read(.username) ?? "Unknown"
let email = AuthenticationCache.read(.email) ?? "Unknown"
Instabug.setUserData("username=\(username) | email=\(email)")
```

**Comment by Kmohamed** [comment]
Hi @Iha-ryq
setUserData only accepting String. And has limit 1000 characters. This is the cause of the crash

**Comment by lha-ryq** [comment]
Hi @Kmohamed 
I don't think it is related with characters limit. This crash happened only once and if it would be related with characters limit then it should happen during each application startup. But of course we will add appropiate check before calling setUserData.

**Comment by Kmohamed** [comment]
I am sorry for misunderstanding 😞
I mean that we use range of string for limiting the usage of setUserData to 1000 characters.
So it must accept string otherwise it will give unrecognized selector with rangeofString method
So the main cause that it must accept string

**Comment by lha-ryq** [comment]
I am not sure I understand you correctly 🤔 
There is no way to pass something else than `String` to `setUserData` method. It wouldn't compile for Swift.

**Comment by lha-ryq** [comment]
We also have similar crash which happened on iOS11.

```
Crash due to signal: SIGSEGV(SEGV_MAPERR) at 6f322beb8

0   libobjc.A.dylib                 (null)
1   Instabug                        -[IBGMethodsLogger LogMethod:] (IBGMethodsLogger.m:71)
2   Instabug                        __24+[Instabug setUserData:]_block_invoke (Instabug.m:156)
3   Foundation                      (null)
4   Foundation                      (null)
5   Foundation                      (null)
6   libdispatch.dylib               (null)
7   libdispatch.dylib               (null)
8   libdispatch.dylib               (null)
9   libdispatch.dylib               (null)
10  libdispatch.dylib               (null)
11  Foundation                      (null)
12  libdispatch.dylib               (null)
13  libdispatch.dylib               (null)
14  libdispatch.dylib               (null)
15  libdispatch.dylib               (null)
16  libdispatch.dylib               (null)
17  libsystem_pthread.dylib         (null)
18  libsystem_pthread.dylib         (null)
```

**Comment by Kmohamed** [comment]
Let me try to reproduce the crash.

**Comment by Kmohamed** [comment]
@lha-ryq The second one is fixed on the latest version.
But to be sure that this fix the issue. you are calling setUserData from thread other than MainThread

**Comment by lha-ryq** [comment]
Thanks @Kmohamed for the information.
We will update Instabug to the newest version.

**Comment by Kmohamed** [comment]
Hi @lha-ryq,
Actually the first crash is also fixed could you please update to SDK version 7.3.5 and inform if you find this crash again.

**Comment by lha-ryq** [comment]
Thanks @Kmohamed  for the investigation!
We have scheduled `Instabug` framework update to the latest version.

**Comment by SudoPlz** [comment]
I got that too:

```
EXC_BAD_ACCESS Attempted to dereference garbage pointer 0x91f1fbec8. 
    /usr/lib/libobjc.A.dylib objc_retain
    Instabug/Logging/IBGDataStore.m:569:9 -[IBGDataStore addEventLogWithName:params:]
    Instabug/Instabug.m:309:9 __40+[Instabug logUserEventWithName:params:]_block_invoke
    Frameworks/Foundation.framework/Foundation ___NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
    Frameworks/Foundation.framework/Foundation -[NSBlockOperation main]
    Frameworks/Foundation.framework/Foundation -[__NSOperationInternal _start:]
    Frameworks/Foundation.framework/Foundation ___NSOQSchedule_f
    /usr/lib/system/libdispatch.dylib __dispatch_client_callout
    /usr/lib/system/libdispatch.dylib __dispatch_queue_serial_drain
    /usr/lib/system/libdispatch.dylib __dispatch_queue_invoke
    /usr/lib/system/libdispatch.dylib __dispatch_root_queue_drain
    /usr/lib/system/libdispatch.dylib __dispatch_worker_thread3
    /usr/lib/system/libsystem_pthread.dylib __pthread_wqthread
```

**Comment by Kmohamed** [comment]
@SudoPlz This issue is fixed on the latest version 7.3.6. Feel free to open it again if you face this problem on this version.
